zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java

```diff
@@ -0,0 +1,1182 @@
+import ohos.agp.render.Canvas;
+import ohos.agp.render.LinearShader;
+import ohos.agp.render.Paint;
+import ohos.agp.render.Shader.TileMode;
+import ohos.agp.utils.Color;
+import ohos.agp.utils.Point;
+import ohos.agp.utils.RectFloat;
+import ohos.app.Context;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+import com.hmos.compat.utils.AttrUtils;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+/**
+ * 水平进度条*
+ * Created by 周卓 on 2016/9/22.
+ */
+
+public class ZzHorizontalProgressBar extends Component implements DrawTask {
+
+
+    private final HiLogLabel hiLogLabel = new HiLogLabel(0, 0, "ZzHorizontalProgressBar");
+    public static final int SHOW_MODE_ROUND = 0;
+    public static final int SHOW_MODE_RECT = 1;
+    public static final int SHOW_MODE_ROUND_RECT = 2;
+
+    public static final int SHAPE_POINT = 0;
+    public static final int SHAPE_LINE = 1;
+
+    private int mMax;
+
+    private int mProgress;
+
+    private int mBgColor;
+
+    private int mProgressColor;
+
+    private int mPadding;
+
+    private boolean mOpenGradient;
+
+    private int mGradientFrom;
+
+    private int mGradientTo;
+
+    private boolean mShowSecondProgress;
+
+    private int mSecondProgress;
+
+    private int mSecondProgressShape;
+
+    private boolean mShowZeroPoint;
+
+    private Paint mSecondProgressPaint;
+    private Paint mSecondGradientPaint;
+    private Paint mProgressPaint;
+    private Paint mGradientPaint;
+    private Paint mBgPaint;
+
+    private boolean mOpenSecondGradient;
+
+    private int mSecondGradientFrom;
+
+    private int mSecondGradientTo;
+
+    private int mSecondProgressColor;
+
+    private int mRadius;
+
+    private boolean mDrawBorder = false;
+
+    private int mBorderColor;
+
+    private int mBorderWidth;
+
+    private int mShowMode = 0;
+
+    private Paint mBorderPaint;
+
+
+    @Target({ElementType.PARAMETER, ElementType.METHOD})
+    @Retention(RetentionPolicy.SOURCE)
+    private @interface ShowMode {
+    }
+
+    @Target({ElementType.PARAMETER, ElementType.METHOD})
+    @Retention(RetentionPolicy.SOURCE)
+    private @interface SecondProgressShape {
+    }
+
+    private OnProgressChangedListener mOnProgressChangedListener;
+
+    /**
+     * OnProgressChangedListener interface.
+     */
+    public interface OnProgressChangedListener {
+        void onProgressChanged(ZzHorizontalProgressBar progressBar, int max, int progress);
+
+        void onSecondProgressChanged(ZzHorizontalProgressBar progressBar, int max, int progress);
+    }
+
+    /**
+     * First Constructor.
+     *
+     * @param context    context as parameter.
+     *
+     */
+    public ZzHorizontalProgressBar(Context context) {
+        this(context, null);
+        init(null);
+    }
+
+    /**
+     * Second Constructor.
+     *
+     * @param context   context as parameter.
+     *
```

Comment:
Line gap is not required between each param

Reply:
done